### PR TITLE
Generate checksums and upload binaries and checksum file to release

### PR DESCRIPTION
**Pull request by hnrkndrssn**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -211,6 +211,8 @@ jobs:
       OCTOPUS_CLI_API_KEY: ${{ secrets.OCTOPUS_API_KEY }}
       VERSION: ${{ needs.build.outputs.version }}
     steps:
+    - uses: actions/checkout@v3
```
Comment (hnrkndrssn) [inline .github/workflows/build.yml:214]: This is needed for `gh cli` to pick up the repository automatically so we don't have to pass it via the command line

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -224,6 +226,26 @@ jobs:
         name: docker-image
         path: artifacts
 
+    - name: Generate sha256 checksums
+      if: github.event_name == 'release'
+      run: |
+        for filePath in $(find  artifacts/* -name '*.tar.gz' -or -name '*.zip')
+        do
+          fileName="$(basename -- $filePath)"
```
Comment (hnrkndrssn) [inline .github/workflows/build.yml:234]: Get the file name without its path

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -224,6 +226,26 @@ jobs:
...
+          newSha=$(sha256sum "$filePath" | awk '{ print $1 }')
```
Comment (hnrkndrssn) [inline .github/workflows/build.yml:235]: Calculate the `sha256` checksum for the file and grab just the checksum (the output from `sha256sum` includes the file path as well)

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -224,6 +226,26 @@ jobs:
...
+          echo "$newSha $fileName" >> artifacts/OctopusTools.${{ needs.build.outputs.version }}.sha256sums
+        done
+
+    - name: Upload binaries and checksum file to release
+      if: github.event_name == 'release'
+      run: |
+        args=("v${{ needs.build.outputs.version }}")
```
Comment (hnrkndrssn) [inline .github/workflows/build.yml:242]: Add the tag for the release to the arguments to pass to `gh`

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -224,6 +226,26 @@ jobs:
...
+        for filePath in $(find  artifacts/* -name '*.tar.gz' -or -name '*.zip' -or -name '*.sha256sums')
+        do 
+          args+=($filePath)
```
Comment (hnrkndrssn) [inline .github/workflows/build.yml:245]: Add each individual file we want to upload to the release to the arguments to pass to `gh` (same as the ones that we calculated checksums for in the previous step)

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -224,6 +226,26 @@ jobs:
         name: docker-image
         path: artifacts
 
+    - name: Generate sha256 checksums
+      if: github.event_name == 'release'
+      run: |
+        for filePath in $(find  artifacts/* -name '*.tar.gz' -or -name '*.zip')
```
Comment (hnrkndrssn) [inline .github/workflows/build.yml:232]: This will include our `win-x64`, `linux-x64`, `osx-x64`, and `portable` binaries

**Review by borland — APPROVED** [review]

**Review by slewis74 — APPROVED** [review]

**Pull request merged**